The qyotoshared is more of a real library than a basic glue library and sholud be treated as such.
This at least builds, but is not runtime tested yet.

Index: b/csharp/qyoto/core/QObjectExtras.cs
===================================================================
--- a/csharp/qyoto/core/QObjectExtras.cs
+++ b/csharp/qyoto/core/QObjectExtras.cs
@@ -21,10 +21,10 @@
 		private static extern void FindQObjectChildren(IntPtr parent, IntPtr regexp,
 									string childName, AddToListFn addFn);
 		
-		[DllImport("libqyotoshared", CharSet=CharSet.Ansi)]
+		[DllImport("libqyoto", CharSet=CharSet.Ansi)]
 		private static extern bool ConnectDelegate(IntPtr obj, string signal, Delegate d, IntPtr handle);
 		
-		[DllImport("libqyotoshared", CharSet=CharSet.Ansi)]
+		[DllImport("libqyoto", CharSet=CharSet.Ansi)]
 		private static extern bool DisconnectDelegate(IntPtr obj, string signal, Delegate d);
 		
 		public static bool Connect(QObject obj, string signal, SlotFunc d) {
@@ -124,4 +124,4 @@
 			return list;
 		}
 	}
-}
\ No newline at end of file
+}
Index: b/csharp/qyoto/src/SmokeMarshallers.cs
===================================================================
--- a/csharp/qyoto/src/SmokeMarshallers.cs
+++ b/csharp/qyoto/src/SmokeMarshallers.cs
@@ -127,7 +127,7 @@
 		[DllImport("libqyoto", CharSet=CharSet.Ansi)]
 		public static extern void InstallInvokeCustomSlot(InvokeCustomSlotFn callback);
 		
-		[DllImport("libqyotoshared", CharSet=CharSet.Ansi)]
+		[DllImport("libqyoto", CharSet=CharSet.Ansi)]
 		public static extern void InstallInvokeDelegate(InvokeDelegateFn callback);
 		
 		[DllImport("libqyoto", CharSet=CharSet.Ansi)]
@@ -1060,4 +1060,4 @@
 #endregion
 
 	}
-}
\ No newline at end of file
+}
